stac_model/schema.py

```diff
@@ -44,12 +44,12 @@ def mlm_prefix_adder(field_name: str) -> str:
 
 class MLModelProperties(Runtime):
     name: str = Field(min_length=1)
-    architecture: str = Field(min_length=1)
+    architecture: Annotated[str | None, OmitIfNone] = Field(min_length=1)
```

Comment:
This one is required. It can be a generic name, but something must be given to identify what the model is. 

https://github.com/stac-extensions/mlm?tab=readme-ov-file#item-properties-and-collection-fields